src/bindings/python/docs/code_examples.md

```diff
@@ -0,0 +1,414 @@
+
+<!-- Pure Python solution describes Python based approach -->
+#### Pure Python solution
+One of the simplest way to extend existing codebase is by writing it in pure Python.
+
+###### Before start: Layout of the project
+How does OpenVINO:tm: packaging work? It is strictly connected to the layout of the Python API itself and reused in different supporting packages like tools and extensions. The main namespace `openvino` provides unified place that connects all packages together during importing, **this is the required part**. However, it is up to the developer how to organize the rest of the package. There are also other common namespaces which follow the same rules:
+* `openvino.tools`
+* ...
+
+For further reading, please refer to: https://packaging.python.org/en/latest/guides/packaging-namespace-packages/
+
+##### Creating new package that extends OpenVINO:tm: project namespace
+Let's go over the example available in `openvino/src/bindings/python/docs/examples/openvino`:
+
+```
+openvino/               <-- Main package/namespace
+    __init__.py         <-- Unified file between all packages
+    mymodule/           <-- This is your new module and it's contents:)
+        __init__.py
+        ...
+        myclass.py
+```
+
+Now let's add it to your exisiting `PYTHONPATH` (replace `[your_path]` with correct path to the OpenVINO:tm: project):
+
+    export PYTHONPATH=$PYTHONPATH:[your_path]/openvino/src/bindings/python/docs/examples/
+
+Test how it works:
+
+```python
+import openvino.mymodule as ov_module
+
+obj = ov_module.MyClass()
+obj.say_hello()
+>>> "Hello! Let's work on OV together!"
+```
+
+##### Extending of existing API (sub)modules
+But how to extend existing API? Let's navigate to `openvino/src/bindings/python/src/openvino` and add something to project helpers. Create new directory and fill it's contents:
+
+```
+openvino/
+    frontend/
+    helpers/                        <-- Working directory
+        __init__.py
+        custom_module/              <-- New directory
+            __init__.py             <-- New file
+            custom_helpers.py       <-- New file
+        packing.py
+    ...
+    runtime/
+    test_utils/
+    __init__.py
+    utils.py
+```
+
+Let's add in `custom_module/custom_helpers.py`:
+```python
+def top1_index(results: list) -> int:
```

Comment:
It could be renamed to `argmax` to make the functions purpose obvious from the start, without looking at its body to make sure. Technically argmax can return a set of points instead of a single index (so it may not be a precise name), so it's up to your decision.

Reply:
I didn't consider `argmax` as a name since it would conflict with numpy function. I would like to keep it as-is.